packages/react-devtools-extensions/src/utils.js

```diff
@@ -2,9 +2,33 @@
 
 import type {BrowserTheme} from 'react-devtools-shared/src/devtools/views/DevTools';
 
-export const IS_EDGE = navigator.userAgent.indexOf('Edg') >= 0;
-export const IS_FIREFOX = navigator.userAgent.indexOf('Firefox') >= 0;
-export const IS_CHROME = IS_EDGE === false && IS_FIREFOX === false;
+export let IS_EDGE;
+export let IS_FIREFOX;
+export let IS_CHROME;
+
+function getBrowserData() {
+  if (navigator.userAgentData) {
+    navigator.userAgentData.brands.forEach(item => {
+      switch (item.brand.toLowerCase()) {
+        case 'google chrome':
+          IS_CHROME = true;
+          break;
+        case 'firefox':
+          IS_FIREFOX = true;
+          break;
+        case 'edge':
+          IS_EDGE = true;
+          break;
+      }
+    });
```

Comment:
```suggestion
    const brandItems = navigator.userAgentData.brands;

    IS_EDGE = brandItems.some(item => item.brand === 'Microsoft Edge');
    IS_FIREFOX = brandItems.some(item => item.brand === 'Mozilla Firefox');
    IS_CHROME = brandItems.some(item => item.brand === 'Google Chrome');
```

Lets put it this way, if there will be some weird custom user agents. This also changes string for Edge to `Microsoft Edge`:
<img width="752" alt="Screenshot 2023-06-23 at 17 09 55" src="https://github.com/facebook/react/assets/28902667/b05f3505-20d6-4ec2-a4c6-5282e298bae8">